packages/ai/core/generate-text/content-part.ts

```diff
@@ -11,4 +11,14 @@ export type ContentPart<TOOLS extends ToolSet> =
   | ({ type: 'source' } & Source)
   | { type: 'file'; file: GeneratedFile } // different because of GeneratedFile object
   | ({ type: 'tool-call' } & ToolCallUnion<TOOLS>)
-  | ({ type: 'tool-result' } & ToolResultUnion<TOOLS>);
+  | ({
+      type: 'tool-result';
+    } & ToolResultUnion<TOOLS>)
+  | {
+      type: 'server-tool-result';
+      toolCallId: string;
+      toolName: string;
+      result: unknown;
+      isError?: boolean;
+      providerMetadata?: ProviderMetadata;
+    };
```

Comment:
why do we need a separate type? can we force an output schema on server side tools and get the generic type that way?